### ISSUE TITLE
Weight Initialization blocks learning

### ISSUE THREAD

**Issue by Giuseppe5** [PyTorchLightning/pytorch-lightning#1465]
## 🐛 Bug

When trying to perform a custom weight initialization, such as xavier_uniform_ or orthogonal_ from torch.nn.init, the weights are kept fixed and not updated during backpropagation.

### To Reproduce

1. Create a simple model, even with just FC layers and ReLU 
2. Initialize the weight with torch.nn.init.xavier_uniform_ or another method
3. Start Training
4. Weights are not getting updated

### Expected behavior

I'd expect the initialization not to block the training process

### Environment

```
packages:
numpy:               1.17.2
pyTorch_debug:       False
pyTorch_version:     1.3.0
pytorch-lightning:   0.7.3
tensorboard:         2.0.0
tqdm:                4.45.0
system:
OS:                  Linux
architecture:
64bit
processor:           x86_64
python:              3.6.9
version:             #163-Ubuntu SMP Mon Sep 24 13:14:43 UTC 2018
```

**Comment by jeffling** [comment]
Do you have any code to share? I couldn't repo

**Comment by awaelchli** [comment]
Your model/algorithm might just be sensitive to the initialization. I just tried the template model from this repo (gpu_template.py) and tried different initializations in the `__init__()` of the template model, e.g.,  
 ```python 
torch.nn.init.uniform_(self.c_d2.weight, -1, 1)
torch.nn.init.uniform_(self.c_d1.weight, -1, 1)
torch.nn.init.uniform_(self.c_d1.bias, -1, 1)
torch.nn.init.uniform_(self.c_d2.bias, -1, 1)
```
and it seems all of the methods I tried are worse than the default one used by PyTorch but if I wait long enough, the loss goes down to zero.
Feel free to re-open if you find evidence that this is a problem with PL.

**Comment by Demirrr** [comment]
Hello @awaelchli,

currently, I am experiencing the same issue described by @Giuseppe5 .

In the context of neural arhictecture search, I would like to increase the embedding vector size by 4 after each epoch ends. 

```python
class KGE(pl.LightningModule):
     ....
def training_epoch_end(self, training_step_outputs):
       ....
       self.embedding_dim += 4
       self.emb_ent_real = nn.Embedding(self.num_entities,self.embedding_dim)
       self.emb_rel_real = nn.Embedding(self.num_relations, self.embedding_dim)
```

During the training process, `self.emb_ent_real` and ` self.emb_rel_real` are not updated, i.e., gradient of the loss w.r.t. parameters (`self.emb_ent_real`, `self.emb_rel_real`) are not used in the paramter update.
